tests/ci/run_analytics.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+set -exo pipefail
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+source tests/ci/common_posix_setup.sh
+
+# The build and test time is not meant to be scientific benchmark, just to get a rough sense of changes overtime.
+# In case there is a weird change over time this recorded additional information about the host to the logs.
+lscpu
+
+function put_metric {
+  # This call to publish the metric could fail but we don't want to fail the build +e turns off exit on error
+  set +e
+  aws cloudwatch put-metric-data \
+    --namespace AWS-LC \
+    "$@" || echo "Publishing metric failed, continuing with the rest of the build"
+  # Turn it back on for the rest of the build
+  set -e
+}
+
+# Return the size of an object or total for the folder (summarize)
+function size {
+  du --bytes --apparent-size --summarize "$@" | cut -f1
+}
+
+SOURCE_CODE_SIZE=$(size "$SRC_ROOT")
+put_metric --metric-name FolderSize --value "$SOURCE_CODE_SIZE" --unit Bytes --dimensions "Folder=root"
+
+for FOLDER_PATH in "$SRC_ROOT"/*/ ; do
+    FOLDER_SIZE=$(size "$FOLDER_PATH")
+    FOLDER=$(basename "$FOLDER_PATH")
+    put_metric --metric-name FolderSize --value "$FOLDER_SIZE" --unit Bytes --dimensions "Folder=${FOLDER}"
+done
+
+function run_build_and_collect_metrics {
+  if [[ "$small" == "ON" ]]; then
+    build_size="Small"
+  else
+    build_size="Large"
+  fi
+  if [[ "$no_assembly" == "ON" ]]; then
+    assembly="NoAsm"
+  else
+    assembly="Asm"
+  fi
+  if [[ "$shared_library" == "ON" ]]; then
+    linking="Shared"
+  else
+    linking="Static"
+  fi
+  if [[ "$fips" == "ON" ]]; then
+    fips_mode="FIPS"
+  else
+    fips_mode="NotFIPS"
+  fi
+  common_dimensions="Optimization=Release,BuildSize=${build_size},Assembly=${assembly},CPU=${PLATFORM},Linking=${linking},FIPS=${fips_mode}"
```

Comment:
I'm not sure if it makes sense, but what about capturing the size of all .obj files?  If we are tracking the size of the build output and want to figure out why it's gotten bigger or smaller then we could slice and dice on the individual. .obj files.